### PR TITLE
Stop stuffing "common" everywhere, shellcheck ftw

### PR DESCRIPTION
**Pull request by evelikov**
The current "common" is fairly misleading - literally half of it is used exclusively within genfstab, plus the in-tree tests. So split it out and use as-needed.

This allows one to reason what is used where - aka `is_array()` has been unused for years - and addresses some shellcheck warnings. A proper editorconfig and gitlab-CI has been copied from mkinitcpio to aid along the way.

This PR builds on top of https://github.com/archlinux/arch-install-scripts/pull/49 and https://github.com/archlinux/arch-install-scripts/pull/47 - mostly to avoid trivial conflicts.

### CODE REVIEW

**Comment by Foxboron** [comment]
The editorconfig should mirror `mkinitcpio` preferably.

https://gitlab.archlinux.org/archlinux/mkinitcpio/mkinitcpio/-/blob/master/.editorconfig

**Comment by evelikov** [comment]
Are you saying that we should re-indent everything? It's a bit annoying (git blame, pending MRs) but can do.

Mindy related - 99.99% of the yml out there use 2 space indent and some linters complain if that's no the case. Any preferences? Fix it here + mkinitcpio, continue with 4 space?

**Comment by Foxboron** [comment]
>git blame, pending MRs

We could possibly do it later. But git has a `.git-blame-ignore-revs` file we can use to ignore the commits.

**Comment by evelikov** [comment]
Yes git has `blame.ignoreRevsFile` although one has to explicitly set it && the filename is non-standard. Also doesn't help with the pending PRs.

**Comment by Foxboron** [comment]
That name is what Github implements and Gitlab has on their todo. So I'd just call it "standard" at this point.

File: common [hunk 1]
```diff
@@ -72,7 +72,7 @@ declare -A fsck_types=([btrfs]=0    # btrfs doesn't need a regular fsck utility
                        [vfat]=1
                        [xfs]=1)
 
-out() { printf "$1 $2\n" "${@:3}"; }
+out() { printf "%s %s\n%s" "$1" "$2" "${@:3}"; }
```
Comment (YHNdnzj) [inline common:1]: This is not applicable as mentioned. We use external format string here.

Comment (evelikov) [reply common:1]: Right - will rebase/fix once there's a bit more feedback && the pre-requirements have landed.

**Comment by Foxboron** [comment]
Rest of your changes are rebased towards #49 which was rebased and the two others where left behind. Preferably just base it off master so I don't need to deal with this directly :)

**Comment by evelikov** [comment]
changes:
  - rebase against master
  - swap printf "fix" for silencing the warning (thanks again for the help)
  - drop msg2 - unused ever since it was introduced
  - drop (unused) msg/warning helpers from fstab-helpers

**Comment by evelikov** [comment]
Humble poke? Is there anything I can do to move this forward - be that as a whole or in part?

**Comment by Foxboron** [comment]
@evelikov Sorry haven't allocated time for arch-install-scripts for a couple of weeks :) I'll try make some time for it soon'ish

File: inline-file.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+sed -e "/$1/ {
+r $3
+d }" <"$2"
```
Comment (Foxboron) [inline inline-file.sh:5]: Any reason why this needs to be a script and can't be inlined in the Makefile?

**Comment by Foxboron** [comment]
Rest looks good to me. Just wondering if we can remove that new script in the root of the project. Duplicating it 3 times in the Makefile wouldn't be so bad.

File: inline-file.sh (hunk 2 above)
Comment (evelikov) [reply inline-file.sh:5]: That was my initial inclination, but didn't manage to make it work. If you have anything that works I'll be happy to respin/integrate.

Comment (evelikov) [reply inline-file.sh]: Ack, removed the m4 -> sed change for now.

For some context, the switch was based on a number of factors. Mentioning some for posterity:
 - declining adoption and knowledge of m4
 - single implementation - gnu m4 vs bsd/gnu sed
 - very infrequent releases - v1.4.19 (2 years ago), .18 (6 years) .17 (10 years)
 - seriously stagnated development - 1.4 branch, ~50 meaningful changes over 10 years, 1.6 branch AWOL for 13 years, 2.0 branch AWOL for 6 years

Comment (Foxboron) [reply inline-file.sh]: Yep, I agree. Maybe we should opt for an `arch-install-scripts` library thing like devtools/mkinitcpio/makepkg does instead.

However that can be a seperate PR i think.

Comment (evelikov) [reply inline-file.sh]: The library approach makes sense when there is notable amount of shared code (and thus duplication) or a test suite doing unit checking individual functionality.

Currently we're at 100 loc and zero tests - so quite far off on both fronts.

Just pushed an update after re-testing everything - genfstab was off :facepalm:

Comment (Foxboron) [reply inline-file.sh]: That's not true, we do have tests.

https://github.com/archlinux/arch-install-scripts/tree/master/test

I don't see why that would be an arbitrary bar to cross though :)

**Comment by Foxboron** [comment]
Thanks, I'll review and merge tonight/this week.

File: inline-file.sh (hunk 2 above)
Comment (evelikov) [reply inline-file.sh]: I feel like we're going off a tangent here?

**Comment by evelikov** [comment]
Thank you :bow:

**Pull request merged**